src/rpc.rs

```diff
@@ -0,0 +1,364 @@
+//! The `rpc` module implements the Solana rpc interface.
+
+use bank::Bank;
+use bs58;
+use jsonrpc_core::*;
+use jsonrpc_http_server::*;
+use request::{Request as JsonRpcRequest, Response};
+use service::Service;
+use signature::{Pubkey, Signature};
+use std::mem;
+use std::net::SocketAddr;
+use std::sync::Arc;
+use std::thread::{self, Builder, JoinHandle};
+
+pub const RPC_PORT: u16 = 8899;
+
+pub struct JsonRpcService {
+    thread_hdl: JoinHandle<()>,
+}
+
+impl JsonRpcService {
+    pub fn new(bank: Arc<Bank>, rpc_addr: SocketAddr) -> Self {
+        let request_processor = JsonRpcRequestProcessor::new(bank);
+        let thread_hdl = Builder::new()
+            .name("solana-jsonrpc".to_string())
+            .spawn(move || {
+                let mut io = MetaIoHandler::default();
+                let rpc = RpcSolImpl;
+                io.extend_with(rpc.to_delegate());
+
+                let server = ServerBuilder::new(io)
+                    .meta_extractor(move |_req: &hyper::Request| Meta {
+                        request_processor: Some(request_processor.clone()),
+                    })
+                    .threads(4)
+                    .cors(DomainsValidation::AllowOnly(vec![
+                        AccessControlAllowOrigin::Any,
+                    ]))
+                    .start_http(&rpc_addr)
+                    .unwrap();
+                server.wait();
+                ()
+            })
+            .unwrap();
+        JsonRpcService { thread_hdl }
+    }
+}
+
+impl Service for JsonRpcService {
+    fn thread_hdls(self) -> Vec<JoinHandle<()>> {
+        vec![self.thread_hdl]
+    }
+
+    fn join(self) -> thread::Result<()> {
+        self.thread_hdl.join()
+    }
+}
+
+#[derive(Clone)]
+pub struct Meta {
+    pub request_processor: Option<JsonRpcRequestProcessor>,
+}
+impl Metadata for Meta {}
+impl Default for Meta {
+    fn default() -> Self {
+        Meta {
+            request_processor: None,
+        }
+    }
+}
+
+build_rpc_trait! {
+    pub trait RpcSol {
+        type Metadata;
+
+        #[rpc(meta, name = "solana_confirmTransaction")]
```

Comment:
Pity to have that `solana_` prefix everywhere. What happens if we got rid of that?

Reply:
Nothing bad. I will do that. We might want to use prefixes at some point to differentiate method types, but there's no reason to clutter it up with an meaningless one now.